src/main/java/uk/gov/hmcts/reform/refunds/services/NotificationServiceImpl.java

```diff
@@ -243,4 +251,21 @@ public Notification getNotificationDetails(MultiValueMap<String, String> headers
         return notificationDetails;
     }
 
+    public String retrieveCustomerReference(MultiValueMap<String, String> headers, String paymentReference) {
+        String customerReference = "";
+        List<String> paymentReferenceList = new ArrayList<>();
+        paymentReferenceList.add(paymentReference);
+
+        PaymentGroupResponse paymentData = paymentService.fetchPaymentGroupResponse(headers, paymentReference);
+
+        // Loop through the payment responses to get the customer reference
+        for (PaymentResponse paymentDtoResponse : paymentData.getPayments()) {
+            if (paymentDtoResponse.getCustomerReference() != null) {
+                customerReference = paymentDtoResponse.getCustomerReference();
+                break;
+            }
+        }
+        return customerReference;
+    }
+
```

Comment:
Olie try to use Stream

Reply:
// Use streams to find the first non-null customer reference
customerReference = paymentData.getPayments().stream()
    .map(PaymentResponse::getCustomerReference)
    .filter(Objects::nonNull)
    .findFirst()
    .orElse(null);